plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/NullPushdownDataTypeTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.clickhouse;
+
+import io.trino.Session;
+import io.trino.spi.type.Type;
+import io.trino.sql.planner.plan.FilterNode;
+import io.trino.sql.query.QueryAssertions;
+import io.trino.sql.query.QueryAssertions.QueryAssert;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.datatype.ColumnSetup;
+import io.trino.testing.datatype.DataSetup;
+import io.trino.testing.sql.TemporaryRelation;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public final class NullPushdownDataTypeTest
+{
+    private final List<TestCase> testCases = new ArrayList<>();
+    private final Set<Integer> specialColumns = new HashSet<>();
+    private final boolean connectorExpressionOnly;
+
+    private NullPushdownDataTypeTest(boolean connectorExpressionOnly)
+    {
+        this.connectorExpressionOnly = connectorExpressionOnly;
+    }
+
+    public static NullPushdownDataTypeTest create()
+    {
+        return new NullPushdownDataTypeTest(false);
+    }
+
+    public static NullPushdownDataTypeTest connectorExpressionOnly()
+    {
+        return new NullPushdownDataTypeTest(true);
+    }
+
+    public NullPushdownDataTypeTest addSpecialColumn(String inputType, String inputLiteral, Type expectedType, String expectedLiteral)
+    {
+        checkArgument(!"NULL".equalsIgnoreCase(inputLiteral), "Special column should not be NULL");
+        specialColumns.add(testCases.size());
+        testCases.add(new TestCase(inputType, inputLiteral, expectedType, expectedLiteral));
```

Comment:
Instead of adding them to same test case - Can we maintain a separate list of test case for special column which we could add them whe we are creating table ?